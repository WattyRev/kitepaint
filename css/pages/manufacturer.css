/**
 * Styles for manufacturer page
 */
/**
 * Base
 * Declare variables and mixins for frontend layout
 */
/* Colors */
/* Mixins */
/* line 7, ../../scss/pages/manufacturer.scss */
#manufacturer {
  font-size: 24px;
  color: #444;
}
/* line 11, ../../scss/pages/manufacturer.scss */
#manufacturer #landing {
  padding: 40px 0;
}
/* line 14, ../../scss/pages/manufacturer.scss */
#manufacturer #landing h2 {
  width: 30%;
  padding: 40px 5% 40px 15%;
  float: left;
}
/* line 19, ../../scss/pages/manufacturer.scss */
#manufacturer #landing form {
  width: 30%;
  padding: 45px 20% 45px 0;
  text-align: center;
  float: right;
}
/* line 25, ../../scss/pages/manufacturer.scss */
#manufacturer #landing .button {
  display: inline-block;
}
