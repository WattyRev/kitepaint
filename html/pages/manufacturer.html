<div id="landing" class="section">
	<div class="content">
		<div class="text">
			<h1>Get started now</h1>
			<p>We will work directly with you to create a custom kite designer for your products.</p>
		</div>
		<form ng-hide="inquired" ng-submit="send()">
			<div>
				<label for="input-name">Name</label>
				<input type="text" id="input-name" placeholder="e.g. Joe Bob" ng-model="input_name" required />
				<label for="input-email">Email</label>
				<input type="email" id="input-email" placeholder="e.g. joe@bob.com" ng-model="input_email" required />
				<label for="input-company">Company</label>
				<input type="text" id="input-company" ng-model="input_company" />
				<button class="button action">Get Started</button>
			</div>
		</form>
		<div class="message" ng-show="inquired">
			<div>
				<h2>We got your request!</h2>
				<p>We will email you soon to follow up on your request to get custom kite design tools for your products.</p>
			</div>
		</div>
		<div class="clearfix"></div>
	</div>
</div>
<div id="about" class="section">
	<div class="content">
		<h1>What can KitePaint do for you?</h1>
		<p>KitePaint.com offers tools for customers to create custom designs for your products. We encourage engagement to keep your customers interested in your products. Here are some of the things that you can get by signing up with KitePaint.com</p>
		<ul>
			<li>Listing on KitePaint.com</li>
			<li>A kite designer web application for your product</li>
			<li>An embeddable designer application to include on your website</li>
		</ul>
	</div>
</div>
<div id="packages" class="section">
	<div class="content">
		<h1>Packages</h1>
		<p>There are three service packages to choose from:</p>
		<table>
			<thead>
				<tr>
					<th></th>
					<th>Basic</th>
					<th>Improved</th>
					<th>Advanced</th>
				</tr>
			</thead>
			<tbody>
				<tr>
					<th>Products <tooltip content="Individualy_listed_product"></tooltip></th>
					<td>1</td>
					<td>5</td>
					<td>Unlimited</td>
				</tr>
				<tr>
					<th>Variations <tooltip content="Additional_variations_of_a_product"></tooltip></th>
					<td>3</td>
					<td>15</td>
					<td>Unlimited</td>
				</tr>
				<tr>
					<th>Embed <tooltip content="Embeddable_version_of_kite_designer"></tooltip></th>
					<td>Yes</td>
					<td>Yes</td>
					<td>Yes</td>
				</tr>
				<tr>
					<th>Price <tooltip content="Prices_are_listed_in_USD"></tooltip></th>
					<td>$20/mo</td>
					<td>$50/mo</td>
					<td>$100/mo</td>
				</tr>
				<tr>
					<td></td>
					<td>
						<button class="button" ng-click="show_lightbox = true; package = 'Basic'">Get Info</button>
					</td>
					<td>
						<button class="button" ng-click="show_lightbox = true; package = 'Improved'">Get Info</button>
					</td>
					<td>
						<button class="button" ng-click="show_lightbox = true; package = 'Advanced'">Get Info</button>
					</td>
				</tr>
			</tbody>
		</table>
		<p><strong>Products</strong> are stand alone items (e.g. Revolution B-Series Pro is a product). A product includes a single variation.</p>
		<p><strong>Variations</strong> are slightly different versions of a product (e.g. Revolution B-Series Pro Vented is a variation). A variation should still be branded as the initial product, and should look very similar with limited variance.</p> 
	</div>
</div>

<div class="lightbox" ng-show="show_lightbox">
	<a ng-click="show_lightbox = false; package = false" class="close right">Close</a>
	<div class="content">
		<h2>Inquire about the {{package}} package</h2>
		<form ng-submit="send(); show_lightbox = false">
			<div>
				<label for="input-name">Name</label>
				<input type="text" id="input-name" placeholder="e.g. Joe Bob" ng-model="input_name" required />
				<label for="input-email">Email</label>
				<input type="email" id="input-email" placeholder="e.g. joe@bob.com" ng-model="input_email" required />
				<label for="input-company">Company</label>
				<input type="text" id="input-company" ng-model="input_company" />
				<button class="button action">Get Started</button>
			</div>
		</form>
	</div>
</div>