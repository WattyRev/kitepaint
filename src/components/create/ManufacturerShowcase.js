import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import manufacturerShape from "../../models/manufacturer";
import { H3, A, P } from "../../theme";

const StyleWrapper = styled.div`
  background: ${props => props.theme.colors.blueLighter};
  border: 1px solid ${props => props.theme.colors.gray};
  margin: 8px 16px;
  border-radius: 4px;
  padding: 8px;
`;

const HeadingWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const StyledImage = styled.div`
  background-image: url(${props => props.src});
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  display: block;
  width: 70px;
  height: 70px;
  margin: 8px 8px 0 0;
  border: 1px solid ${props => props.theme.colors.gray};
  border-radius: 4px;
  box-sizing: border-box;
  background-color: ${props => props.theme.colors.white};
`;

const ManufacturerShowcase = ({ manufacturer, children }) => (
  <StyleWrapper>
    <HeadingWrapper>
      <StyledImage src={`/logos/${manufacturer.logo}`} />
      <div>
        <H3>{manufacturer.name}</H3>
        {manufacturer.website && (
          <P>
            <A href={manufacturer.website} target="_blank">
              {manufacturer.website}
            </A>
          </P>
        )}
      </div>
    </HeadingWrapper>
    {children}
  </StyleWrapper>
);

ManufacturerShowcase.propTypes = {
  manufacturer: manufacturerShape,
  children: PropTypes.node
};

export default ManufacturerShowcase;
