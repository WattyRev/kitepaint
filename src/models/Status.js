import PropTypes from "prop-types";

/**
 * Status can be applied to any model to indicate it's public visibility.
 */
export const statusProp = PropTypes.oneOf(["0", "1", "2"]);
const Status = {
  "0": "Private",
  "1": "Unlisted",
  "2": "Public",
  PRIVATE: "0",
  UNLISTED: "1",
  PUBLIC: "2"
};
export default Status;
