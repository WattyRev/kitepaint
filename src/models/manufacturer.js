import PropTypes from "prop-types";

/**
 * A manufacturer is a company that manufactures kites and owns one or more products.
 */
const manufacturerState = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  logo: PropTypes.string.isRequired,
  website: PropTypes.string
});

export default manufacturerState;

/**
 * A mock manufacturer used for testing
 * @type {Object}
 */
const getMockManufacturer = () => ({
  id: "abc",
  name: "Krazy Kites",
  logo: "abc.jpg",
  website: "http://krazykites.com"
});
export { getMockManufacturer };
