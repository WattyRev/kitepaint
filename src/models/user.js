import PropTypes from "prop-types";

/**
 * A user is an individual user account. It contains limited details that are available publically.
 */
const userShape = PropTypes.shape({
  username: PropTypes.string.isRequired,
  loginid: PropTypes.string.isRequired
});

export default userShape;

/**
 * A mock user used for testing.
 */
const getMockUser = () => ({
  username: "FlyFucker",
  loginid: "abc"
});

export { getMockUser };
