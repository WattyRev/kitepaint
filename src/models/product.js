import PropTypes from "prop-types";
import { getMockManufacturer } from "./manufacturer";
import Status, { statusProp } from "./status";

/**
 * A product is a kite that can be customized to create a design.
 */
const productShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  embed: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  notes: PropTypes.arrayOf(PropTypes.string),
  manufacturer: PropTypes.string.isRequired,
  status: statusProp.isRequired,
  url: PropTypes.string,
  colors: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      color: PropTypes.string.isRequired
    })
  ),
  variations: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      svg: PropTypes.string.isRequired
    })
  )
});

export default productShape;

/**
 * A mock product used for testing
 * @type {Object}
 */
const getMockProduct = () => ({
  id: "product-1",
  embed: "krazykites.com",
  name: "Krazy Kite",
  manufacturer: getMockManufacturer().id,
  url: "http://krazykites.com/krazy-kite",
  colors: [
    {
      name: "red",
      color: "#ff0000"
    },
    {
      name: "black",
      color: "#000000"
    }
  ],
  variations: [
    {
      name: "Standard",
      svg: "<div>Kool Kite</div>"
    }
  ],
  status: Status.PUBLIC
});

export { getMockProduct };
