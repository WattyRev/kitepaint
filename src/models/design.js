import PropTypes from "prop-types";
import { getMockProduct } from "./product";
import Status, { statusProp } from "./status";

/**
 * A design is a user-created entity. It represetns a saved design of a particular product.
 */
const designShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  // A string representing the created date
  created: PropTypes.string.isRequired,
  // A string representing the updated date
  updated: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  // The ID of the cooresponding product
  product: PropTypes.string.isRequired,
  // The ID of the user that created the design
  user: PropTypes.string,
  variations: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      svg: PropTypes.string.isRequired,
      primary: PropTypes.bool.isRequired
    })
  ),
  // The status of the cooresponding product
  productStatus: statusProp.isRequired,
  // The status of the design
  status: statusProp.isRequired
});

export default designShape;

/**
 * A mock design used for testing
 * @type {Object}
 */
const getMockDesign = () => ({
  id: "design-1",
  created: "01/01/2018",
  updated: "01/02/2018",
  name: "picasso",
  product: getMockProduct().id,
  variations: [
    {
      name: "Standard",
      svg: "<div>picasso standard</div>",
      primary: true
    }
  ],
  status: Status.PRIVATE,
  productStatus: Status.PUBLIC
});
export { getMockDesign };
