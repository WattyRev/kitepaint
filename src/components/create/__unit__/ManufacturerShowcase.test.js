import React from "react";
import { shallow, mount } from "enzyme";
import Theme from "../../../theme";
import { getMockManufacturer } from "../../../models/manufacturer";
import ManufacturerShowcase, { StyleWrapper } from "../ManufacturerShowcase";

describe("ManufacturerShowcase", () => {
  let mockProps;
  beforeEach(() => {
    mockProps = {
      manufacturer: getMockManufacturer()
    };
  });
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(
      <ManufacturerShowcase {...mockProps}>
        <div>Children!</div>
      </ManufacturerShowcase>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders the website if provided", () => {
    expect.assertions(1);
    mockProps.manufacturer.website = "http://test.com";
    const wrapper = shallow(
      <ManufacturerShowcase {...mockProps}>
        <div>Children!</div>
      </ManufacturerShowcase>
    );
    expect(wrapper.find(".testing_website")).toHaveLength(1);
  });
  it("does not render the website if not provided", () => {
    expect.assertions(1);
    mockProps.manufacturer.website = "";
    const wrapper = shallow(
      <ManufacturerShowcase {...mockProps}>
        <div>Children!</div>
      </ManufacturerShowcase>
    );
    expect(wrapper.find(".testing_website")).toHaveLength(0);
  });

  describe("StyleWrapper", () => {
    it("renders", () => {
      expect.assertions(1);
      const wrapper = mount(<StyleWrapper theme={Theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
