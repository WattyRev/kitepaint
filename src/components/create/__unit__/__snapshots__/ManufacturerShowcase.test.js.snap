// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManufacturerShowcase StyleWrapper renders 1`] = `
.c0 {
  background: #f0fbff;
  border: 1px solid #AAAAAA;
  margin: 8px 16px;
  border-radius: 4px;
  padding: 8px;
}

<styled.div
  theme={
    Object {
      "colors": Object {
        "aqua": "#7FDBFF",
        "black": "#111111",
        "blue": "#0074D9",
        "blueDarker": "#005daa",
        "blueLighter": "#f0fbff",
        "fuchsia": "#F012BE",
        "gray": "#AAAAAA",
        "grayDark": "#5e5e5e",
        "grayDarker": "#3a3a3a",
        "green": "#2ECC40",
        "lime": "#01FF70",
        "maroon": "#85144B",
        "navy": "#001F3F",
        "olive": "#3D9970",
        "orange": "#FF851B",
        "purple": "#B10DC9",
        "red": "#FF4136",
        "silver": "#DDDDDD",
        "teal": "#39CCCC",
        "white": "#FFFFFF",
        "yellow": "#FFDC00",
      },
      "patterns": Object {
        "transparencyBackground": Array [
          "
  background-image: linear-gradient(45deg, ",
          "#AAAAAA",
          " 25%, transparent 25%),
    linear-gradient(-45deg, ",
          "#AAAAAA",
          " 25%, transparent 25%),
    linear-gradient(45deg, transparent 75%, ",
          "#AAAAAA",
          " 75%),
    linear-gradient(-45deg, transparent 75%, ",
          "#AAAAAA",
          " 75%);
  background-color: ",
          "#5e5e5e",
          ";
  background-size: 20px 20px;
  background-position: 0 0, 0 10px, 10px -10px, -10px 0px;
",
        ],
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-kAzzGY",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  background: ",
            [Function],
            ";
  border: 1px solid ",
            [Function],
            ";
  margin: 8px 16px;
  border-radius: 4px;
  padding: 8px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-kAzzGY",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "colors": Object {
          "aqua": "#7FDBFF",
          "black": "#111111",
          "blue": "#0074D9",
          "blueDarker": "#005daa",
          "blueLighter": "#f0fbff",
          "fuchsia": "#F012BE",
          "gray": "#AAAAAA",
          "grayDark": "#5e5e5e",
          "grayDarker": "#3a3a3a",
          "green": "#2ECC40",
          "lime": "#01FF70",
          "maroon": "#85144B",
          "navy": "#001F3F",
          "olive": "#3D9970",
          "orange": "#FF851B",
          "purple": "#B10DC9",
          "red": "#FF4136",
          "silver": "#DDDDDD",
          "teal": "#39CCCC",
          "white": "#FFFFFF",
          "yellow": "#FFDC00",
        },
        "patterns": Object {
          "transparencyBackground": Array [
            "
  background-image: linear-gradient(45deg, ",
            "#AAAAAA",
            " 25%, transparent 25%),
    linear-gradient(-45deg, ",
            "#AAAAAA",
            " 25%, transparent 25%),
    linear-gradient(45deg, transparent 75%, ",
            "#AAAAAA",
            " 75%),
    linear-gradient(-45deg, transparent 75%, ",
            "#AAAAAA",
            " 75%);
  background-color: ",
            "#5e5e5e",
            ";
  background-size: 20px 20px;
  background-position: 0 0, 0 10px, 10px -10px, -10px 0px;
",
          ],
        },
      }
    }
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`ManufacturerShowcase renders 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      src="/logos/abc.jpg"
    />
    <div>
      <Styled(H3)>
        Krazy Kites
      </Styled(H3)>
      <P
        className="testing_website"
      >
        <styled.a
          href="http://krazykites.com"
          target="_blank"
        >
          http://krazykites.com
        </styled.a>
      </P>
    </div>
  </styled.div>
  <div>
    Children!
  </div>
</styled.div>
`;
